### PR TITLE
Added ATS Probes and bpftrace script Checking for Idle Connections

### PR DESCRIPTION
**Pull request by BronzyPlum6390**
Added ATS Probes and BPF trace script for counting the number of connections. This checks for idle connections. I am having an issue for my bpftrace not supporting the interval for loop which is why I am using a while loop.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
     "version": "0.2.0",
     "configurations": [
+        
+
         {
             "name": "(gdb) traffic_server",
             "type": "cppdbg",
             "request": "launch",
-            "program": "${workspaceFolder}/${env:ATS_VSCODE_BUILDDIR}/src/traffic_server/traffic_server",
+            "program": "/tmp/ats-dev/bin/traffic_server",
```
Comment (JosiahWI) [inline .vscode/launch.json:10]: Why are you hardcoding this? Could this not ever be installed in a different directory?

**Review by JosiahWI — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (BronzyPlum6390) [reply .vscode/launch.json]: Resolved

**Review by BronzyPlum6390 — COMMENTED** [review]
Fixed hardcoding issue

**Comment by JosiahWI** [comment]
Thanks for fixing it.

**Comment by JosiahWI** [comment]
Did you mean to close this @BronzyPlum6390?

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -42,5 +42,5 @@ endif()
 add_library(catch2::catch2 INTERFACE IMPORTED GLOBAL)
 target_include_directories(catch2::catch2 INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/catch2")
 
-add_library(systemtap::systemtap INTERFACE IMPORTED GLOBAL)
-target_include_directories(systemtap::systemtap INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/systemtap")
+#add_library(systemtap::systemtap INTERFACE IMPORTED GLOBAL)
+#target_include_directories(systemtap::systemtap INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/systemtap")
```
Comment (bneradt) [inline lib/CMakeLists.txt:46]: Why are these being removed? We build this locally so as to remove a dependency upon systemtap.

@moonchen ^

File: tests/gold_tests/ats_probe/ats_probe.bt [hunk 3]
```diff
@@ -23,13 +23,83 @@
...
+    printf("\nConnections by client IP:\n");
+    print(@connections);
+    printf("\nTotal lines read: %d\n", @readlines);
+    printf("\nLast transaction stamps by client IP:\n");
+    print(@lasttransaction);
+    printf("\nMin conn_id: %d\n", @min_conn_id);
+    printf("\nMax conn_id: %d\n", @max_conn_id);
+    
 }
+
```
Comment (bneradt) [inline tests/gold_tests/ats_probe/ats_probe.bt:105]: Set your code editor to add trailing newlines at the end of your files.

**Pull request closed without merging**